flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliStatementSplitter.java

```diff
@@ -49,7 +50,10 @@ public static boolean isStatementComplete(String statement) {
         for (String line : content.split("\n")) {
             if (isEndOfStatement(line)) {
                 buffer.add(line);
-                statements.add(String.join("\n", buffer));
+                statements.add(
+                        buffer.stream()
```

Comment:
Please also take a look at line 63.

Reply:
Could you filter the empty line after `map` operation, e.g.

```
buffer.stream()
  .map(...)
  .filter(...) // only keep the non-empty line here
  .collect(...)
```

Reply:
Thanks , I will fix these problems